tycho-core/src/main/java/org/eclipse/tycho/core/TargetPlatformConfiguration.java

```diff
@@ -136,11 +135,8 @@ public String getTargetPlatformResolver() {
     }
 
     public synchronized List<TargetDefinitionFile> getTargets() {
-        for (Iterator<Supplier<File>> iterator = lazyTargetFiles.iterator(); iterator.hasNext();) {
-            Supplier<File> supplier = iterator.next();
-            targets.add(supplier.get().toURI());
-            iterator.remove();
-        }
+        lazyTargetFiles.forEach(supplier -> targets.add(supplier.get().toURI()));
```

Comment:
This is not equivalent, assume loading one lazy target fails (and the method is called multiple times) this will add target definitions multiple times, while before the lazy one is removed as soon as it was successfully queried.

Reply:
Indeed, makes sense.
I thought about using the more modern `removeIf()`, but unfortunately the `removeIf()` method in the default implementation in Collection and in `ArrayList` behave differently in case an exception is thrown (i.e. the former has removed all matches before and the latter just does nothing effectively since the actual removal is postponed to the end).
Thus I'll just keep it as it is.